ultraplot/axes/plot.py

```diff
@@ -3666,6 +3726,134 @@ def fill_betweenx(self, *args, **kwargs):
+
+            xs = [x for x, y in pos.values()]
+            ys = [y for x, y in pos.values()]
+
+            min_x, max_x = min(xs), max(xs)
+            min_y, max_y = min(ys), max(ys)
+
+            width = max_x - min_x
+            height = max_y - min_y
+            pos = {
+                k: (
+                    (x - min_x) / width if width else 0.5,
+                    (y - min_y) / height if height else 0.5,
+                )
+                for k, (x, y) in pos.items()
+            }
+        # Set a sensible default if not given
+        if "node_size" not in node_kw:
+            coords = np.array(list(pos.values()))
+            xlim = self.get_xlim()
+            ylim = self.get_ylim()
+
+            # Size of data space shown
+            data_span = np.array([xlim[1] - xlim[0], ylim[1] - ylim[0]])
+            axis_bbox = self.get_window_extent().transformed(
+                self.figure.dpi_scale_trans.inverted()
+            )
+            axis_size_inch = np.array([axis_bbox.width, axis_bbox.height])
+            dpi = self.figure.dpi
+            axis_size_px = axis_size_inch * dpi
+
+            # Convert a fixed pixel diameter  into data units
+            desired_px_diameter = 7  # px
+            data_units_per_px = data_span / axis_size_px
+            data_diameter = np.mean(data_units_per_px) * desired_px_diameter
+
+            # Convert to `node_size` in pt² (as required by nx.draw)
+            # 1 point = 1/72 inch → diameter in points = (desired_px / dpi) * 72
+            diameter_inch = desired_px_diameter / dpi
+            diameter_pt = diameter_inch * 72
+            radius_pt = diameter_pt / 2
+            node_size = np.pi * radius_pt**2  # area in pt²
+            node_kw["node_size"] = node_size
+
+        # By default soften the edge alpha to prevent "hairball " effect
+        if "alpha" not in edge_kw:
+            n_edges = g.number_of_edges()
+            n = g.number_of_nodes()
+            # For more edges reduce the alpha for the edges
+            alpha = 1 - np.exp(-n_edges / (n - 1))
+            edge_kw["alpha"] = max(
+                alpha, 0.1
+            )  # with a cutt-off to prevent dissapearence for complete graphs
+
+        # Draw the graph using networks functions
+        if nodes:
+            if np.iterable(nodes):
+                node_kw["nodelist"] = nodes
+            nodes = nx.draw_networkx_nodes(g, pos=pos, ax=self, **node_kw)
+        if edges:
+            if np.iterable(edges):
+                edge_kw["edgelist"] = edges
+            edges = nx.draw_networkx_edges(g, pos=pos, ax=self, **edge_kw)
+        if labels:
+            if np.iterable(labels):
+                label_kw["labels"] = labels
+            labels = nx.draw_networkx_labels(g, pos=pos, ax=self, **label_kw)
+
+        # Apply styling
+        self.set_aspect(aspect)
+        self.grid(grid)
+        self.set_facecolor(facecolor)
+        self._toggle_spines(spines)
```

Comment:
We should apply the default styling not via keywords to the function but rather through the standard way ultraplot gathers defaults and possible overrides. We should also provide rcparams.